pkg/plugin/delete_pvc_action_plugin.go

```diff
@@ -66,7 +75,30 @@ func (p *NewPVCDeleteItemAction) Execute(input *velero.DeleteItemActionExecuteIn
 	}
 
 	snapshotID := itemSnapshot.Status.SnapshotID
-	backupRepository := snapshotUtils.NewBackupRepository(itemSnapshot.Spec.BackupRepository)
+	bslName := input.Backup.Spec.StorageLocation
+
+	var backupRepositoryName string
+	isLocalMode := utils.GetBool(install.DefaultBackupDriverImageLocalMode, false)
+	if !isLocalMode {
+		p.Log.Info("Claiming backup repository during delete")
+
+		repositoryParameters := make(map[string]string)
+
+		err = utils.RetrieveParamsFromBSL(repositoryParameters, bslName, restConfig, p.Log)
+		if err != nil {
+			p.Log.Errorf("Failed to translate BSL to repository parameters: %v", err)
+			return errors.WithStack(err)
+		}
+
+		backupRepositoryName, err = backupdriver.ClaimBackupRepository(ctx, utils.S3RepositoryDriver, repositoryParameters,
+			[]string{pvc.Namespace}, veleroNs, backupdriverClient, p.Log)
+		if err != nil {
+			p.Log.Errorf("Failed to claim backup repository: %v", err)
+			return errors.WithStack(err)
+		}
+	}
+	backupRepository := snapshotUtils.NewBackupRepository(backupRepositoryName)
```

Comment:
Can we think about wrap this piece of code snippet around a utility function, like, NewBackupRepositoryFromBSLName? So, all of our three ItemAction plugins can consume on it without maintaining three replicated code pieces.